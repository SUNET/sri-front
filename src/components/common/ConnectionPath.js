import React from 'react';

import '../../style/PortConnectionPath.scss';

const ConnectionPath = ({ blocks }) => {
  return (
    <div className="connection-path">
      {blocks.map((el, index, arr) => {
        const isLast = index === arr.length - 1;
        return (
          <div key={`connection-path__key-${index}`} className="connection-path__element">
            <div className="connection-path__element__content">
              <div className="connection-path__element__content__link-name">
                {<a
                  href={`${el.path}`}
                  rel="noopener noreferrer"
                  onClick={(e) => {
                    e.stopPropagation();
                  }}
                >
                  {el.name}
                </a>}
                {el.portName && !el.currentElement && (
                  <a
                    href={`${el.portPath}`}
                    rel="noopener noreferrer"
                    onClick={(e) => {
                      e.stopPropagation();
                    }}
                  >
                    {el.portName}
                  </a>
                )}
                {el.portName && el.currentElement && el.portName}
              </div>
              <div className="connection-path__element__content__type">({el.connectionType})</div>
            </div>
            {!isLast && <div className="connection-path__element__arrow"></div>}
          </div>
        );
      })}
    </div>
  );
};

export default ConnectionPath;
