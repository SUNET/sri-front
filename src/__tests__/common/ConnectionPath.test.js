import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ConnectionPath from '../../components/common/ConnectionPath';

let ConnectionPathWrapper;
let ConnectionPathComponent;
let InputElement;

const defaultProps = {
  changeConnectionPath: jest.fn(),
  t: jest.fn((t) => t),
  blocks: [
        {
            id: 'fake-id1',
            name: 'test1450',
            connectionType: 'C13 / C14',
            path: '/network/routers/fake-id1',
            portName: '14500',
            portPath: '/network/ports/UG9ydDo3MDE1',
            currentElement: true,
        },
        {
            id: 'fake-id2',
            name: '14400-14500',
            connectionType: 'Dark Fiber',
            path: '/network/cables/fake-id2',
        },
        {
            id: 'fake-id3',
            name: 'test1440',
            connectionType: 'C13 / C14',
            path: '/network/external-equipments/fake-id3',
            portName: '14400',
            portPath: '/network/ports/UG9ydDo3MDEy',
        },
    ]
};

const CreateConnectionPathWrapper = (props) => {
  const allProps = { ...defaultProps, ...props };
  return (
    <div>
      <ConnectionPath {...allProps} />
    </div>
  );
};

const mountConnectionPathWrapper = (props) => mount(<CreateConnectionPathWrapper {...props} />);

beforeEach(() => {
  jest.clearAllMocks();
  ConnectionPathWrapper = mountConnectionPathWrapper();
  ConnectionPathComponent = ConnectionPathWrapper.find('ConnectionPath');
  InputElement = ConnectionPathComponent.find('input');
});
afterEach(() => {
  ConnectionPathWrapper.unmount();
});

describe('Filter Component', () => {
  it('Renders correctly', () => {
    expect(toJson(ConnectionPathWrapper)).toMatchSnapshot();
    expect(ConnectionPathComponent.find('.connection-path')).toHaveLength(1);
  });
});
