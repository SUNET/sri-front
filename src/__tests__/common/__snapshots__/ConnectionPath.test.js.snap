// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component Renders correctly 1`] = `
<CreateConnectionPathWrapper>
  <div>
    <ConnectionPath
      blocks={
        Array [
          Object {
            "connectionType": "C13 / C14",
            "currentElement": true,
            "id": "fake-id1",
            "name": "test1450",
            "path": "/network/routers/fake-id1",
            "portName": "14500",
            "portPath": "/network/ports/UG9ydDo3MDE1",
          },
          Object {
            "connectionType": "Dark Fiber",
            "id": "fake-id2",
            "name": "14400-14500",
            "path": "/network/cables/fake-id2",
          },
          Object {
            "connectionType": "C13 / C14",
            "id": "fake-id3",
            "name": "test1440",
            "path": "/network/external-equipments/fake-id3",
            "portName": "14400",
            "portPath": "/network/ports/UG9ydDo3MDEy",
          },
        ]
      }
      changeConnectionPath={[MockFunction]}
      t={[MockFunction]}
    >
      <div
        className="connection-path"
      >
        <div
          className="connection-path__element"
          key="connection-path__key-0"
        >
          <div
            className="connection-path__element__content"
          >
            <div
              className="connection-path__element__content__link-name"
            >
              <a
                href="/network/routers/fake-id1"
                onClick={[Function]}
                rel="noopener noreferrer"
              >
                test1450
              </a>
              14500
            </div>
            <div
              className="connection-path__element__content__type"
            >
              (
              C13 / C14
              )
            </div>
          </div>
          <div
            className="connection-path__element__arrow"
          />
        </div>
        <div
          className="connection-path__element"
          key="connection-path__key-1"
        >
          <div
            className="connection-path__element__content"
          >
            <div
              className="connection-path__element__content__link-name"
            >
              <a
                href="/network/cables/fake-id2"
                onClick={[Function]}
                rel="noopener noreferrer"
              >
                14400-14500
              </a>
            </div>
            <div
              className="connection-path__element__content__type"
            >
              (
              Dark Fiber
              )
            </div>
          </div>
          <div
            className="connection-path__element__arrow"
          />
        </div>
        <div
          className="connection-path__element"
          key="connection-path__key-2"
        >
          <div
            className="connection-path__element__content"
          >
            <div
              className="connection-path__element__content__link-name"
            >
              <a
                href="/network/external-equipments/fake-id3"
                onClick={[Function]}
                rel="noopener noreferrer"
              >
                test1440
              </a>
              <a
                href="/network/ports/UG9ydDo3MDEy"
                onClick={[Function]}
                rel="noopener noreferrer"
              >
                14400
              </a>
            </div>
            <div
              className="connection-path__element__content__type"
            >
              (
              C13 / C14
              )
            </div>
          </div>
        </div>
      </div>
    </ConnectionPath>
  </div>
</CreateConnectionPathWrapper>
`;
